repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -3362,38 +3341,67 @@ public void deleteClassificationOnlyPropagation(Set<String> deletedEdgeIds) thro
         RequestContext.get().getDeletedEdgesIds().addAll(deletedEdgeIds);
 
         for (AtlasEdge edge : deletedEdgeIds.stream().map(x -> graph.getEdge(x)).collect(Collectors.toList())) {
-            Map<AtlasVertex, List<AtlasVertex>> removePropagationsMap  = deleteDelegate.getHandler().getRemovePropagationMap(edge);
 
-            if (removePropagationsMap == null) {
+            boolean isRelationshipEdge = deleteDelegate.getHandler().isRelationshipEdge(edge);
+            boolean isTermEntityEdge = isTermEntityEdge(edge);
+
+            if (edge == null || !isRelationshipEdge) {
```

Comment:
isTermEntityEdge check should be here instead of line 3361.

Reply:
It should be there because isTermEntityEdge is not mandatory case for edge so that we can abort the operation or continue